.travis.yml

```diff
@@ -0,0 +1,232 @@
+sudo: required
+
+language: generic
+
+stages:
+  - setup
+  - compile
+  - name: buildrpm
+  - test
+  - coverage
+  - name: docs
+    if: ( ( branch = master ) OR ( tag =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc|alpha|dev)([0-9]+$))?|(-git[0-9a-z]{6,8})?$ ) )
+  - name: prerelease
+    if: tag =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc)([0-9]+))$
+  - name: devrelease
+    if: tag =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc)([0-9]+))$
+  - name: release
+    if: tag =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)$
+  - name: deployment
+    if: tag =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc|alpha|dev)([0-9]+$))?|(-git[0-9a-z]{6,8})?$
+
+.docs: &docs_cond
+  if: ( ( branch = master ) OR ( tag =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc|alpha|dev)([0-9]+$))?|(-git[0-9a-z]{6,8})?$ ) )
+.draftrelease: &draftrelease_cond
+  condition: $TRAVIS_TAG =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc|dev|alpha)([0-9]+))?(-git[0-9a-z]{6,8})$
+.prerelease:   &prerelease_cond
+  condition: $TRAVIS_TAG =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc)([0-9]+))$
+.devrelease:   &devrelease_cond
+  condition: $TRAVIS_TAG =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(dev|alpha)([0-9]+))$
+.release:      &release_cond
+  condition: $TRAVIS_TAG =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)$
+# .deploy:       &deploy_cond
+#   condition: $TRAVIS_TAG =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc|alpha|dev)([0-9]+$))?|(-git[0-9a-z]{6,8})?$
+.deploy:       &deploy_cond
+  if: tag =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc|alpha|dev)([0-9]+$))?|(-git[0-9a-z]{6,8})?$
+
+env:
+  global:
+    ## Encrypted variables:
+    ##  * CHANGELOG_GITHUB_TOKEN: for running github_changelog_generator without API rate limits
+    # - secure: i4K316XPS0ANZeU6bF0JIsxZOaD/7gD5IwXR+kQwbENLpf6DsrLTE6ZJKgS/r9jClOAvbAQj9R7O61uRCHPYzQemKkXhCLSlqMlk0/RdvLVlcFCR6C7SDPF5DKnA8QQ3r7c2jdv27/TFfk60VWOYbQln94Mh8n3kH+6NGOwFwjxLBce88v8lsZXb3NDiHzwxXMeTbqqrAAbhCrjr7hrIvpm7eJwXVhOCz2HR7SpDOIFhwaDgGNgotqleF2szXwVnOUE4RxOgR7zBKLLFwjOtcdS0nRvOctf7l6kiq0t16V+BRIr0kad9hMUscNEKHDmQmRWF9zI83k2j/Ux4kgq1NUy6Jx+fbs/Y0WIwrd+LCyGIuaw/Y+Y3LVwT1EmUzYoga7oq2VB7j07PrOvGa4BwFDN1zm+7n2EZtmDw5aZwTtMeCkm4L+dL8HoiVVWAaFMZ8l/EYDDqWea5dcg9uazdI/wUFLavuc1lRzGTGHK+fxEYmX5DOfK35nyJQqDjB6dCKFjsty9NR1nF1wnsk49IuvviN6IfpCiIaGTxgAUrV8mT32gxPB9DdDyiPsQXHS6Z8ha2aq9Eeq7Zm/TXA5WkbUcw3RJsmdX6oDj89FzElX5EK0KjwRJgqgFfDP7yIhQHsRSk0QnWhIO6YI78A30DaTwR7vKNex09eT3igm+4ppU=
```

Comment:
Should the github token be available in public?

Reply:
This is the secure way to do things, the string is encrypted, and can only be decrypted in certain circumstances